### ISSUE TITLE
Custom Column not working on 0.36.5.1

### ISSUE THREAD

**Issue by flamber** [metabase/metabase#13241]
**Describe the bug**
Custom Column not working on 0.36.5.x

**To Reproduce**
1. Custom question > Sample Dataset > Orders
2. Custom Column formula `1 + 1` as "Test" - formula can be anything, with or without math or a column
![image](https://user-images.githubusercontent.com/1447303/93017658-bd848900-f5ca-11ea-9335-f13d32f3ff41.png)
3. Errors with `Must be a valid instance of one of these clauses: :field-id, :field-literal`

<details><summary>Full stacktrace</summary>

```
09-13 14:06:26 ERROR middleware.catch-exceptions :: Error processing query: null
{:database_id 4,
 :started_at #t "2020-09-13T14:06:25.736084+02:00[Europe/Copenhagen]",
 :error_type :invalid-query,
 :json_query
 {:database 4, :query {:source-table 11, :expressions {:Test ["field-id" 88]}}, :type "query", :parameters [], :middleware {:js-int-to-string? true, :add-default-userland-constraints? true}},
 :native nil,
 :status :failed,
 :class clojure.lang.ExceptionInfo,
 :stacktrace
 ["--> mbql.util$fn__25448$unwrap_field_clause__25453.invoke(util.clj:430)"
  "mbql.util$fn__25507$field_clause__GT_id_or_literal__25512$fn__25513.invoke(util.clj:459)"
  "mbql.util$fn__25507$field_clause__GT_id_or_literal__25512.invoke(util.clj:449)"
  "query_processor.middleware.large_int_id$convert_id_to_string$fn__45596$fn__45599.invoke(large_int_id.clj:36)"
  "query_processor.middleware.large_int_id$convert_id_to_string$fn__45596.invoke(large_int_id.clj:44)"
  "query_processor.middleware.limit$limit$fn__45621.invoke(limit.clj:38)"
  "query_processor.middleware.format_rows$format_rows$fn__45576.invoke(format_rows.clj:81)"
  "query_processor.middleware.desugar$desugar$fn__45050.invoke(desugar.clj:22)"
  "query_processor.middleware.binning$update_binning_strategy$fn__44076.invoke(binning.clj:229)"
  "query_processor.middleware.resolve_fields$resolve_fields$fn__44592.invoke(resolve_fields.clj:24)"
  "query_processor.middleware.add_dimension_projections$add_remapping$fn__38811.invoke(add_dimension_projections.clj:311)"
  "query_processor.middleware.add_implicit_clauses$add_implicit_clauses$fn__39018.invoke(add_implicit_clauses.clj:141)"
  "query_processor.middleware.add_source_metadata$add_source_metadata_for_source_queries$fn__39411.invoke(add_source_metadata.clj:105)"
  "query_processor.middleware.reconcile_breakout_and_order_by_bucketing$reconcile_breakout_and_order_by_bucketing$fn__46515.invoke(reconcile_breakout_and_order_by_bucketing.clj:98)"
  "query_processor.middleware.auto_bucket_datetimes$auto_bucket_datetimes$fn__43717.invoke(auto_bucket_datetimes.clj:125)"
  "query_processor.middleware.resolve_source_table$resolve_source_tables$fn__44639.invoke(resolve_source_table.clj:46)"
  "query_processor.middleware.parameters$substitute_parameters$fn__46300.invoke(parameters.clj:114)"
  "query_processor.middleware.resolve_referenced$resolve_referenced_card_resources$fn__44691.invoke(resolve_referenced.clj:80)"
  "query_processor.middleware.expand_macros$expand_macros$fn__45306.invoke(expand_macros.clj:158)"
  "query_processor.middleware.add_timezone_info$add_timezone_info$fn__39442.invoke(add_timezone_info.clj:15)"
  "query_processor.middleware.splice_params_in_response$splice_params_in_response$fn__47212.invoke(splice_params_in_response.clj:32)"
  "query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__46526$fn__46530.invoke(resolve_database_and_driver.clj:33)"
  "driver$do_with_driver.invokeStatic(driver.clj:61)"
  "driver$do_with_driver.invoke(driver.clj:57)"
  "query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__46526.invoke(resolve_database_and_driver.clj:27)"
  "query_processor.middleware.fetch_source_query$resolve_card_id_source_tables$fn__45524.invoke(fetch_source_query.clj:267)"
  "query_processor.middleware.store$initialize_store$fn__47221$fn__47222.invoke(store.clj:11)"
  "query_processor.store$do_with_store.invokeStatic(store.clj:46)"
  "query_processor.store$do_with_store.invoke(store.clj:40)"
  "query_processor.middleware.store$initialize_store$fn__47221.invoke(store.clj:10)"
  "query_processor.middleware.cache$maybe_return_cached_results$fn__44568.invoke(cache.clj:209)"
  "query_processor.middleware.validate$validate_query$fn__47230.invoke(validate.clj:10)"
  "query_processor.middleware.normalize_query$normalize$fn__45648.invoke(normalize_query.clj:22)"
  "query_processor.middleware.add_rows_truncated$add_rows_truncated$fn__39280.invoke(add_rows_truncated.clj:36)"
  "query_processor.middleware.results_metadata$record_and_return_metadata_BANG_$fn__47197.invoke(results_metadata.clj:147)"
  "query_processor.middleware.constraints$add_default_userland_constraints$fn__44927.invoke(constraints.clj:42)"
  "query_processor.middleware.process_userland_query$process_userland_query$fn__46389.invoke(process_userland_query.clj:136)"
  "query_processor.middleware.catch_exceptions$catch_exceptions$fn__44870.invoke(catch_exceptions.clj:174)"
  "query_processor.reducible$async_qp$qp_STAR___38074$thunk__38075.invoke(reducible.clj:101)"
  "query_processor.reducible$async_qp$qp_STAR___38074.invoke(reducible.clj:107)"
  "query_processor.reducible$sync_qp$qp_STAR___38083$fn__38086.invoke(reducible.clj:133)"
  "query_processor.reducible$sync_qp$qp_STAR___38083.invoke(reducible.clj:132)"
  "query_processor$process_userland_query.invokeStatic(query_processor.clj:215)"
  "query_processor$process_userland_query.doInvoke(query_processor.clj:211)"
  "query_processor$fn__47372$process_query_and_save_execution_BANG___47381$fn__47384.invoke(query_processor.clj:227)"
  "query_processor$fn__47372$process_query_and_save_execution_BANG___47381.invoke(query_processor.clj:219)"
  "query_processor$fn__47416$process_query_and_save_with_max_results_constraints_BANG___47425$fn__47428.invoke(query_processor.clj:239)"
  "query_processor$fn__47416$process_query_and_save_with_max_results_constraints_BANG___47425.invoke(query_processor.clj:232)"
  "api.dataset$fn__50707$fn__50710.invoke(dataset.clj:55)"
  "query_processor.streaming$streaming_response_STAR_$fn__35496$fn__35497.invoke(streaming.clj:73)"
  "query_processor.streaming$streaming_response_STAR_$fn__35496.invoke(streaming.clj:72)"
  "async.streaming_response$do_f_STAR_.invokeStatic(streaming_response.clj:66)"
  "async.streaming_response$do_f_STAR_.invoke(streaming_response.clj:64)"
  "async.streaming_response$do_f_async$fn__23282.invoke(streaming_response.clj:85)"],
 :context :ad-hoc,
 :error "Must be a valid instance of one of these clauses: :field-id, :field-literal",
 :row_count 0,
 :running_time 0,
 :preprocessed nil,
 :ex-data {:type :schema.core/error, :value nil, :error (named (not (some-matching-condition? nil)) "Must be a valid instance of one of these clauses: :field-id, :field-literal")},
 :data {:rows [], :cols []}}

09-13 14:06:26 DEBUG middleware.log :: POST /api/dataset 202 [ASYNC: completed] 295.9 ms (22 DB calls) App DB connections: 0/15 Jetty threads: 3/50 (1 idle, 0 queued) (152 total active threads) Queries in flight: 1 (0 queued)
```

</details>

**Information about your Metabase Installation:**
Metabase 0.36.5 and 0.36.5.1

**Additional context**
https://discourse.metabase.com/t/custom-columns-error-in-version-0-36-5-1-possible-bug/11574

**Comment by sbelak** [comment]
The underlying cause fixed with https://github.com/metabase/metabase/pull/13242. It's not ideal that we have a flag (`qp.middleware.large-int-id/js-int-to-string?`) that's set for most queries as they are run by the user, but not while running tests.

**Comment by dacort** [comment]
FYI, this breaks existing questions that have custom columns as well.

**Comment by flamber** [comment]
Fixed by #13242 - will be in release 0.36.6
